index.js

```diff
@@ -0,0 +1,426 @@
+					if (
+						req.body.password ===
+						AES.decrypt(rows[0].password, `${process.env.AES_key}`)
+					) {
+						req.session.user = rows[0];
+						res.redirect("/dashboard");
+					} else {
+						console.log("Invalid password");
+						res.status(401).render("error", {
+							errorMessage: "Authentication error!"
+						});
+					}
+				} else {
+					console.log("Username not found!");
+					res.status(401).render("error", {
+						errorMessage: "Authentication error!"
+					});
+				}
+			} else {
+				console.log("Username not found!");
+				res.status(401).render("error", {
+					errorMessage: "Authentication error!"
+				});
+			}
+		}
+	);
+});
+app.post("/checkAvailability", (req, res) => {
+	if (req.session.user) {
+		con.query(
+			"SELECT slotHall FROM booking WHERE eventID IN (SELECT id FROM events WHERE dateWanted = ?)",
+			req.body.dateWanted,
+			(err, result) => {
+				if (!err) {
+					res.send(result);
+				} else {
+					console.log(err);
+					res.status(500).render("error", {
+						errorMessage: "Availability checking failed!"
+					});
+				}
+			}
+		);
+	} else {
+		res.redirect("/login");
+	}
+});
+app.post("/makeRequest", (req, res) => {
+	if (req.session.user) {
+		var slotHall = "";
+		for (var i = 1; i <= 7; i++) {
+			for (var j = 1; j <= 9; j++) {
+				if (req.body[`cell${j}${i}`]) {
+					slotHall += j.toString() + i.toString();
+				}
+			}
+		}
+		var eventPayload = {
+			title: req.body.event_name,
+			club: req.body.clubName,
+			requirements: req.body.requirements,
+			details: req.body.desc,
+			dateWanted: req.body.dateWanted,
+			userID: req.session.user.id
+		};
+		con.query("insert into events set ?", eventPayload, (err) => {
+			if (!err) {
+				console.log(`Inserted ${eventPayload.title} into events.`);
+				con.query(
+					"select id from events WHERE userID= ? ORDER BY id DESC LIMIT 1",
+					req.session.user.id,
+					(err, eventID) => {
+						if (!err) {
+							var bookingPayload = {
+								userID: req.session.user.id,
+								eventID: eventID[0].id,
+								slotHall
+							};
+							con.query(
+								"insert into booking set ?",
+								bookingPayload,
+								(err) => {
+									if (!err) {
+										console.log(
+											`Inserted event with id ${bookingPayload.eventID} into booking.`
+										);
+										res.redirect("/dashboard");
+									} else {
+										console.log(err);
+										res.status(500).render("error", {
+											errorMessage:
+												"Inserting into booking failed!"
+										});
+									}
+								}
+							);
+						} else {
+							console.log(err);
+							res.status(500).render("error", {
+								errorMessage: "Retrieving eventID failed!"
+							});
+						}
+					}
+				);
+			} else {
+				console.log(err);
+				res.status(500).render("error", {
+					errorMessage: "Inserting into event failed!"
+				});
+			}
+		});
+	} else {
+		res.redirect("/login");
+	}
+});
+app.post("/onRequestUpdate", (req, res) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Arrow function has a complexity of 5. (complexity)](https://app.codacy.com/app/AravindVNair99/Hall-Management-System/pullRequest?prid=4592231)